From baf581631f863d15be82d8c6550083e61ac91d03 Mon Sep 17 00:00:00 2001
From: Luiz Capitulino <lcapitulino@redhat.com>
Date: Fri, 14 Nov 2014 16:20:26 -0500
Subject: [virt] virtio_balloon: update_balloon_size() - update correct field

Message-id: <20141114112026.58c1e466@redhat.com>
Patchwork-id: 100052
O-Subject: [RHEL7.1 PATCH] virtio_balloon: update_balloon_size(): update correct field
Bugzilla: 1163567
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>
RH-Acked-by: Amit Shah <amit.shah@redhat.com>
RH-Acked-by: Rafael Aquini <aquini@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1163567
Brew-build: http://brewweb.devel.redhat.com/brew/taskinfo?taskID=8244218
Upstream-status: Merged.

According to the virtio spec, the device configuration field
that should be updated after an inflation or deflation
operation is the 'actual' field, not the 'num_pages' one.

Commit 855e0c5288177bcb193f6f6316952d2490478e1c swapped them
in update_balloon_size(). Fix it.

Signed-off-by: Luiz Capitulino <lcapitulino@redhat.com>
Signed-off-by: Rusty Russell <rusty@rustcorp.com.au>
Fixes: 855e0c5288177bcb193f6f6316952d2490478e1c
(cherry picked from commit 3459f11a8b16f40f9cde8e4281c2d5dd2ff1a732)
Signed-off-by: Luiz Capitulino <lcapitulino@redhat.com>

RHEL-NOTE: The bug this commit fixes is actually a high
priority regression in the balloon driver: without this
commit, the balloon driver in the guest doesn't update
QEMU on the current balloon size, so QEMU and other
higher level stack components, like libvirt, will never
know when/if a balloon operation has been completed.
Sorry for not mentioning this in the original commit log.

On RHEL, this bug was backported by commit
a19d3f2b2d964c0b1e7d63e268d1d9cb8930d01b.

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/virtio/virtio_balloon.c b/drivers/virtio/virtio_balloon.c
index 90e567a..ae93caa 100644
--- a/drivers/virtio/virtio_balloon.c
+++ b/drivers/virtio/virtio_balloon.c
@@ -284,7 +284,7 @@ static void update_balloon_size(struct virtio_balloon *vb)
 {
  __le32 actual = cpu_to_le32(vb->num_pages);
 
- virtio_cwrite(vb->vdev, struct virtio_balloon_config, num_pages,
+ virtio_cwrite(vb->vdev, struct virtio_balloon_config, actual,
         &actual);
 }
 
-- 
1.7.1